.github/general_coding_instructions.md

```diff
@@ -0,0 +1,798 @@
+        }
+    }
+    
+callback_will_come:
+    return result;
+}
+```
+
+## Variable Naming Conventions {#variable-naming}
+
+### General Rules
+- Use **lowercase with underscores** (snake_case) for all variable names
+- **NEVER use camelCase, PascalCase, or any mixed-case naming**
+- Use descriptive names that clearly indicate purpose
+- Avoid single-letter variables except for loop counters (`i`, `j`, `k`)
+
+### Specific Patterns
+```c
+// Local variables
+int result = 0;
+THANDLE(BSDL_ADDRESS_LIST) temp = NULL;
+uint32_t record_count = 0;
+bool is_success = false;
+
+// Function parameters
+BSDL_FILE_HL_HANDLE handle
+const char* full_file_name
+CONSTBUFFER_HANDLE user_metadata
+void* user_context
+
+// Structure members
+typedef struct EXAMPLE_TAG
+{
+    uint32_t item_count;
+    BSDL_ADDRESS_HANDLE* address_list;
+    volatile_atomic int32_t ref_count;
+} EXAMPLE;
+```
+
+### Special Variable Types
+- **Handles**: Use `*_handle` suffix for handle variables
+- **Callbacks**: Use `*_callback` suffix for callback function pointers
+- **Contexts**: Use `*_context` suffix for context parameters
+- **Results**: Always use `result` for function return values
+- **Temporaries**: Use `temp` or `tmp_*` prefix for temporary variables
+
+## Result Variable Conventions {#result-variables}
+
+### Initialization
+- **Avoid initializing result variables when possible** to ensure all code paths explicitly set a value:
+```c
+// Preferred - uninitialized to catch missing assignments
+int result;
+TYPE_HANDLE result;
+TYPE_RESULT result;
+
+// EXCEPTION: THANDLE types must always be initialized to NULL
+THANDLE(TYPE) result = NULL;  // Required for reference-counted types
+
+// Only initialize others when necessary (e.g., when goto jumps over assignments)
+int result = MU_FAILURE;  // when using goto that skips the main logic
+```
+
+### Ensuring All Paths Set Results
+- **Every code path must explicitly set the result variable**
+- **THANDLE types are an exception** - they must be initialized to NULL for proper reference counting
+- Use compiler warnings (`-Wuninitialized`) to catch missing assignments for non-THANDLE types
+- This pattern helps identify bugs where error paths don't set return values
+
+### Return Patterns
+- Use consistent return value patterns:
+  - `0` for success, non-zero for failure (integer functions)
+  - `NULL` for failure (pointer/handle functions)
+  - Specific enum values for typed results
+- **Every code path must explicitly assign the result variable**
+- **THANDLE types must be initialized to NULL** due to reference counting requirements
+- Use uninitialized variables and compiler warnings to catch missing assignments for non-THANDLE types
+
+### Return Value Handling
+- **All function return values must be checked or explicitly ignored**
+- Use `(void)` cast to explicitly indicate intentional ignoring of return values:
+
+```c
+// Correct - checking return value
+int result = some_function();
+if (result != 0)
```

Comment:
`if (some_function() != 0) { LogError(); result=MU_FAILURE; }` is better

Reply:
Let's optimize this subsequently!